lib/compat.php

```diff
@@ -472,3 +472,111 @@ function gutenberg_render_block_with_assigned_block_context( $pre_render, $parse
  * @see WP_Block::render
  */
 remove_action( 'enqueue_block_assets', 'wp_enqueue_registered_block_scripts_and_styles' );
+
+/**
+ * Shim that hooks into `wp_update_nav_menu_item` and makes it so that nav menu
+ * items support a 'content' field. This field contains HTML and is used by nav
+ * menu items with `type` set to `'html'`.
+ *
+ * Specifically, this shim makes it so that:
+ *
+ * 1) The `wp_update_nav_menu_item()` function supports setting
+ * `'menu-item-content'` on a menu item. When merged to Core, this functionality
+ * should exist in `wp_update_nav_menu_item()`.
+ *
+ * 2) The `customize_save` ajax action supports setting `'content'` on a nav
+ * menu item. When merged to Core, this functionality should exist in
+ * `WP_Customize_Manager::save()`.
+ *
+ * This shim can be removed when the Gutenberg plugin requires a WordPress
+ * version that has the ticket below.
+ *
+ * @see <TICKET>
+ *
+ * @param int   $menu_id         ID of the updated menu.
+ * @param int   $menu_item_db_id ID of the new menu item.
+ * @param array $args            An array of arguments used to update/add the menu item.
+ */
+function gutenberg_update_nav_menu_item_content( $menu_id, $menu_item_db_id, $args ) {
+	global $wp_customize;
+
+	// Support setting content in customize_save admin-ajax.php requests by
+	// grabbing the unsanitized $_POST values.
+	if ( isset( $wp_customize ) ) {
+		$values = $wp_customize->unsanitized_post_values();
+		if ( isset( $values[ "nav_menu_item[$menu_item_db_id]" ]['content'] ) ) {
+			if ( is_string( $values[ "nav_menu_item[$menu_item_db_id]" ]['content'] ) ) {
+				$args['menu-item-content'] = $values[ "nav_menu_item[$menu_item_db_id]" ]['content'];
+			} elseif ( isset( $values[ "nav_menu_item[$menu_item_db_id]" ]['content']['raw'] ) ) {
+				$args['menu-item-content'] = $values[ "nav_menu_item[$menu_item_db_id]" ]['content']['raw'];
+			}
+		}
+	}
+
+	$defaults = array(
+		'menu-item-content' => '',
+	);
+
+	$args = wp_parse_args( $args, $defaults );
+
+	update_post_meta( $menu_item_db_id, '_menu_item_content', $args['menu-item-content'] );
```

Comment:
`update_post_meta` passes `$args['menu-item-content']` through `wp_unslash`, which means using italics on the label and POSTing contents like `<!-- wp:search {"label":"\u003cem\u003eTest\u003c/em\u003e","placeholder":"test","buttonText":"Search"} /-->` will actually save `<!-- wp:search {"label":"u003cemu003eTestu003c/emu003e","placeholder":"test","buttonText":"Search"} /-->` which in turn breaks the rendering. Passing it through `wp_slash` first as a workaround fixes it. 

```suggestion
	update_post_meta( $menu_item_db_id, '_menu_item_content', wp_slash( $args['menu-item-content'] ) );
```

Reply:
TIL! What a nightmare—thanks!